src/platform/graphics/egl_extensions.cpp

```diff
@@ -56,30 +45,37 @@ std::experimental::optional<mg::EGLExtensions::PlatformBaseEXT> maybe_platform_b
 }
 
 mg::EGLExtensions::EGLExtensions() :
+    platform_base{maybe_platform_base_ext()}
+{
+}
+
+mg::EGLExtensions::BaseExtensions::BaseExtensions(EGLDisplay dpy) :
     eglCreateImageKHR{
         reinterpret_cast<PFNEGLCREATEIMAGEKHRPROC>(eglGetProcAddress("eglCreateImageKHR"))},
     eglDestroyImageKHR{
         reinterpret_cast<PFNEGLDESTROYIMAGEKHRPROC>(eglGetProcAddress("eglDestroyImageKHR"))},
-
     /*
      * TODO: Find a non-ES GL equivalent for glEGLImageTargetTexture2DOES
      * It's the LAST remaining ES-specific function. Although Mesa lets you use
      * it in full GL, it theoretically should not work. Mesa just lets you
      * mix ES and GL code. But other drivers won't be so lenient.
      */
     glEGLImageTargetTexture2DOES{
-        reinterpret_cast<PFNGLEGLIMAGETARGETTEXTURE2DOESPROC>(eglGetProcAddress("glEGLImageTargetTexture2DOES"))},
-    wayland{maybe_wayland_ext()},
-    platform_base{maybe_platform_base_ext()}
+        reinterpret_cast<PFNGLEGLIMAGETARGETTEXTURE2DOESPROC>(eglGetProcAddress("glEGLImageTargetTexture2DOES"))}
 {
-    if (!eglCreateImageKHR || !eglDestroyImageKHR)
-        BOOST_THROW_EXCEPTION(std::runtime_error("EGL implementation doesn't support EGLImage"));
+    auto const egl_extensions = eglQueryString(dpy, EGL_EXTENSIONS);
+    if (!egl_extensions || !strstr(egl_extensions, "EGL_KHR_image_base") || !eglCreateImageKHR || !eglDestroyImageKHR)
+    {
+        BOOST_THROW_EXCEPTION(std::runtime_error("EGL display doesn't support EGL_KHR_image_base"));
+    }
 
-    if (!glEGLImageTargetTexture2DOES)
+    if (!egl_extensions || !strstr(egl_extensions, "GL_OES_EGL_image") || !glEGLImageTargetTexture2DOES)
```

Comment:
Have you tested this? Because I'm pretty sure this will fail, as `GL_OES_EGL_image` is a *GL* extension, not an EGL one.

Reply:
Fixed